### DISCUSSION TITLE
Working with polar coordinates

### DISCUSSION THREAD

**Discussion by mkuemmel** [shapely/shapely#1716] [Q&A]
We are using shapely to work in polar coordinate (RA/Dec) on the entire sky, mostly with polygons. The analogy would be working with longitude and latitude over the entire earth. We do have problems working at the line RA=360 which is also RA=0.0, e.g. checking for polygon overlaps and so on. The documentation says "no support for coordinate transformation" and there are always transformation from 3D to 2D. Is there a specific trick how to work around longitude=0.0?

**Comment by brendan-ward** [comment]
Provided that your overall analysis allows having coordinates outside [0,360], one trick is to find polygons that touch or overlap 0 or 360, and shift them to one side or the other.

So, for example with longitude, I've used the left side of a bounding box compared to a reference longitude (e.g., `0`) to find all geometries I want to move to the left.  This is a gross simplification for my use case because I've already filtered out any cases where values to the right of that reference should be wrapped.  You'll need to do filtering first to find those things that should be wrapped and make sure that is done consistently for both sides of whatever sets of geometries are being used in your overlap analysis (e.g., if one set of geometries has big polygons and another is points or small polygons, need to make sure any that could overlap are uniformly shifted).

Then once you know them, you can just add or subtract out 360 depending on which way you want to go, then union the parts back together.  Not shown here is first splitting any multi-part geometries via `get_parts` before transform, and then `union` of them back together, because those steps are specific to your data structure.


```python
bounds = shapely.bounds(geometries)
out_geometries = geometries.copy()
ix = bounds[:, 0] >= ref_long

if ix.sum():
    # subtract 360 degrees
    out_geometries[ix] = shapely.transform(geometries[ix], lambda g: g - [360, 0])
```

You can also just find any polygons that touch or overlap 0 or 360, duplicate them, and shift one set of them to the left or right of [0,360] domain before doing overlap analysis - but then you'll have to deduplicate after.
